tests/integration/targets/ec2_vpc_dhcp_option/tasks/main.yml

```diff
@@ -0,0 +1,863 @@
+---
+- module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default('') }}"
+      region: "{{ aws_region }}"
+  vars:
+    vpc_cidr: '10.{{ 256 | random(seed=resource_prefix) }}.0.0/24'
+  block:
+
+  - name: create a VPC with a default DHCP option set to test inheritance and delete_old
+    ec2_vpc_net:
+      name: "{{ resource_prefix }}"
+      cidr_block: "{{ vpc_cidr }}"
+      state: present
+    register: vpc
+
+  - set_fact:
+      vpc_id: "{{ vpc.vpc.id }}"
+
+  - name: set up initial DHCP option set
+    environment:
+      AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+      AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+      AWS_SESSION_TOKEN: "{{ security_token | default('') }}"
+      AWS_DEFAULT_REGION: "{{ aws_region }}"
+    block:
+      - name: find all DHCP options sets
+        command: 'aws ec2 describe-dhcp-options --filters --query "DhcpOptions[*].DhcpOptionsId" --output text'
+        register: associations
+
+      - name: find the associated VPCs
+        command: 'aws ec2 describe-vpcs --filters "Name=dhcp-options-id,Values={{ item }}" --query "Vpcs[*].VpcId" --output text'
```

Comment:
Why describe-vpcs rather than ec2_vpc_dhcp_option_info or ec2_vpc_net_info ?

Reply:
FYI, this is a super rough-draft and probably going to change a lot. There's a bunch of stuff that I did because it was quicker than looking up docs. I was just was hoping to see if it was passing on another account yet, since it seems like there are some quirks.